tests/parser/functions/test_as_num256.py

```diff
@@ -0,0 +1,21 @@
+import pytest
+from tests.setup_transaction_tests import chain as s, tester as t, ethereum_utils as u, check_gas, \
+    get_contract_with_gas_estimation, get_contract
+
+def test_as_num256_with_negative_num(assert_tx_failed):
+    code = """
+@public
+def foo() -> num256:
+    return as_num256(1-2)
+"""
+    assert_tx_failed(lambda: get_contract_with_gas_estimation(code), Exception)
```

Comment:
This doesn't have a custom Exception?

Reply:
Nope, it doesn't

Reply:
Would it be possible to let this be a different kind of exception versus the standard TransitionFailed exception?

Reply:
I switched it to compile failed.  Eventually I think we should create custom exceptions for all of `compile_lll.py`

Reply:
For sure, at least something that differentiates lll exceptions from viper (or solidity) ones